pkg/engine/lifecycletest/provider_test.go

```diff
@@ -48,7 +48,7 @@ func TestSingleResourceDefaultProviderLifecycle(t *testing.T) {
 
 	// We should have started the provider 10 times, twice for each of the steps in the basic lifecycle (one preview,
```

Comment:
Comment needs updating, and I think we should TODO this that ideally we don't want to be starting up the provider for a schema when we've already started it for resource control.

Reply:
Good catch, will add the TODO